### PR TITLE
[C-2970] Add Link, Improve Text

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds generic Link component which will be used across the app to improve seo
Improves Text component 
- Adds `as` prop to customize underlying rendered element.
- Updates color prop to use the same color config as Icon and HarmonyButton

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2970-add-link

File: packages/stems/src/styles/colors.ts [hunk 1]
```diff
@@ -38,3 +38,4 @@ export type ColorValue =
   | 'accentPurple'
   | 'accentBlue'
   | 'specialLightGreen'
+  | 'darkmodeStaticWhite'
```
Comment (raymondjacobson) [inline packages/stems/src/styles/colors.ts:41]: staticWhite was supposed to be the same in light/dark, should it be in here?

**Review by raymondjacobson — APPROVED** [review]

File: packages/stems/src/styles/colors.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/stems/src/styles/colors.ts:41]: staticWhite should also be in here eventually, this add was for compatibility with the new banner color add

File: packages/web/src/components/typography/Text.tsx [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
 
-  const Tag: ElementType = variantTagMap[variant][size] ?? 'p'
+  const Tag: ElementType = as ?? variantTagMap[variant][size] ?? 'p'
 
-  const themeColors = getCurrentThemeColors()
-  const textColor = themeColors[color] ?? themeColors['--neutral']
-  const styleObject: CSSProperties = {
-    color: textColor
+  const styleObject: CSSCustomProperties = {
+    '--text-color': `var(${toCSSVariableName(color)})`
```
Comment (Kyle-Shanks) [inline packages/web/src/components/typography/Text.tsx:46]: we should probably fall back to inherit here in case a color is not passed

**Review by Kyle-Shanks — APPROVED** [review]
Nice

**Pull request merged**